src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -21,15 +21,31 @@ public static class TextLoaderSaverCatalog
         /// <param name="catalog">The <see cref="DataOperationsCatalog"/> catalog.</param>
         /// <param name="columns">Array of columns <see cref="TextLoader.Column"/> defining the schema.</param>
         /// <param name="separatorChar">The character used as separator between data points in a row. By default the tab character is used as separator.</param>
-        /// <param name="hasHeader">Whether the file has a header with feature names. Note: If a TextLoader is created with hasHeader = true but without a
-        /// <paramref name="dataSample"/>, then vector columns made by TextLoader will not contain slot name annotations (slots being the elements of the given vector column),
-        /// because the output schema is made when the TextLoader is made, and not when <see cref="TextLoader.Load(IMultiStreamSource)"/> is called.
-        /// In addition, the case where dataSample = null and hasHeader = true indicates to the loader that when it is given a file when Load()
-        /// is called, it needs to skip the first line.</param>
-        /// <param name="dataSample">The optional location of a data sample. The sample can be used to infer column names and number of slots in each column.</param>
-        /// <param name="allowQuoting">Whether the file can contain columns defined by a quoted string.</param>
-        /// <param name="trimWhitespace">Remove trailing whitespace from lines</param>
-        /// <param name="allowSparse">Whether the file can contain numerical vectors in sparse format.</param>
+        /// <param name="hasHeader">Whether the file has a header with feature names. When <see langword="true"/>, the loader will skip the first line when
```

Comment:
>param name="hasHeader" [](start = 13, length = 22)

When a dataSample is provided, then "true" indicates that the first line in the dataSample will be used for feature names, and that when Load() is called the first line will be skipped. When there is no dataSample provided, then "true" just indicates that the loader should skip the first line when Load() is called, but columns will not have slot names annotations. (here you can add the explanation about the slot names being defined in the constructor and not at Load() time.)